internal/comparetest/metrics.go

```diff
@@ -239,14 +251,18 @@ func CompareNumberDataPointSlices(expected, actual pmetric.NumberDataPointSlice)
 // CompareNumberDataPoints compares each part of two given NumberDataPoints and returns
 // an error if they don't match. The error describes what didn't match.
 func CompareNumberDataPoints(expected, actual pmetric.NumberDataPoint) error {
-	if expected.ValueType() != actual.ValueType() {
-		return fmt.Errorf("metric datapoint types don't match: expected type: %s, actual type: %s", numberTypeToString(expected.ValueType()), numberTypeToString(actual.ValueType()))
+	exp, act := pmetric.NewNumberDataPoint(), pmetric.NewNumberDataPoint()
+	expected.CopyTo(exp)
+	actual.CopyTo(act)
```

Comment:
It's not needed in functions that don't mutate the data like this one. Mutating operation in other functions is sorting

Reply:
Got it.